### ISSUE TITLE
Linux support?

### ISSUE THREAD

**Issue by vktRus** [kulikan/privkey#2]
Ноw to compile on Linux (Ubuntu 64bit)?

**Comment by norguhtar** [comment]
- check openssl version on your system. If it 1.1.1 install openssl headers in your system.
- download and unpack 1.1.0.3 release from this page https://github.com/gost-engine/engine/releases in directory with privkey.c
- create build.sh with this content
```
gcc -I/usr/include -Iengine-1.1.0.3 privkey.c engine-1.1.0.3/gost89.c engine-1.1.0.3/gosthash.c \
engine-1.1.0.3/gost_ameth.c engine-1.1.0.3/gost_ctl.c engine-1.1.0.3/gost_asn1.c \
engine-1.1.0.3/gost_ec_sign.c engine-1.1.0.3/e_gost_err.c engine-1.1.0.3/gost_params.c \
engine-1.1.0.3/gosthash2012.c -lcrypto -o privkey
```
- make build.sh executable
- run ./build.sh
...
- PROFIT

**Comment by kulikan** [comment]
Thx Anatoliy!

**Comment by DiamondMax** [comment]
@norguhtar 
Compilation terminated with error
`privkey.c:11:10: fatal error: applink.c: No such file or directory`
If I comment out this line compilation finishes successfully but exporting key always returns "Error check public key"
Ubuntu 20.04, OpenSSL 1.1.1f, headers installed

**Comment by norguhtar** [comment]
> @norguhtar Compilation terminated with error `privkey.c:11:10: fatal error: applink.c: No such file or directory` If I comment out this line compilation finishes successfully but exporting key always returns "Error check public key" Ubuntu 20.04, OpenSSL 1.1.1f, headers installed

Check this https://github.com/yhirose/cpp-httplib/issues/592#issuecomment-667056999
It should be helped

**Comment by DiamondMax** [comment]
> Check this [yhirose/cpp-httplib#592 (comment)](https://github.com/yhirose/cpp-httplib/issues/592#issuecomment-667056999)
> It should be helped

No, that thread is about compilation on Windows. This file applink.c is required and included only for windows version of OpenSSL. 

Там обсуждается компиляция под Windows. Вообще этот файл нужен только для Windows-версии OpenSSL. В Linux-сборку он не входит (об этом пишут в той же ветке). Я пробовал собирать без него, файл компилируется, но работает не корректно: всегда возвращает "Error check public key", хотя пин-код от контейнера ввожу правильный (под виндой из того же контейнера извлекается без ошибок).
